src/domain/bitcoin/lightning/index.types.d.ts

```diff
@@ -6,7 +6,9 @@ type RouteNotFoundError = import("./errors").RouteNotFoundError
 type InvoiceExpiration = Date & { readonly brand: unique symbol }
 type EncodedPaymentRequest = string & { readonly brand: unique symbol }
 type PaymentHash = string & { readonly brand: unique symbol }
-type PaymentSecret = string & { readonly brand: unique symbol }
+type SecretPreImage = string & { readonly brand: unique symbol }
+type RevealedPreImage = string & { readonly brand: unique symbol }
+type PaymentIdentifyingSecret = string & { readonly brand: unique symbol }
```

Comment:
here is another option for naming :

- PaymentPreimage
- PaymentSecret
- PaymentSecretRevealed

Reply:
Do we want to attach this to the `Payment` concept given that we have explicit `LnPayment` types that are different from `LnInvoice` types and both types an carry a preimage?

Reply:
oh you're right. a reason to separate those types further? 😬

Reply:
> oh you're right. a reason to separate those types further? grimacing

I was thinking that as well, but that probably introduces unnecessary complexity.

For example, if in the future we wanted to do hodl invoice tricks and we want to for e.g. generate the preimage+hash for an LnPayment (see [these scenarios](https://bolt.fun/guide/invoices/hodl-invoice#scenarios-source--more-examples)), we could still fulfill this use-case with just `secret`/`revealed` types without having to figure out whether it's attached to an invoice we generate or a payment we make, or some in-between concept of those two.